compiler/rustc_parse_format/src/lib.rs

```diff
@@ -224,7 +237,7 @@ impl<'a> Iterator for Parser<'a> {
                 '{' => {
                     let curr_last_brace = self.last_opening_brace;
                     let byte_pos = self.to_span_index(pos);
-                    let lbrace_end = self.to_span_index(pos + 1);
+                    let lbrace_end = InnerOffset(byte_pos.0 + self.to_span_width(pos));
```

Comment:
Could we wrap all creations of `InnerOffset`?

Reply:
Hmm, not exactly sure what you mean.

Maybe related: a function that gives you both position and width could clean this up